### DISCUSSION TITLE
NODE_ENV must not be forced.

### DISCUSSION THREAD

**Discussion by yahao87** [quasarframework/quasar#12956] [Ideas / Proposals] [answered]
https://github.com/quasarframework/quasar/blob/v1/app/lib/quasar-conf-file.js#L769-L780

I upgraded the Quasar version to 1.13.1 > 1.18.6.
And faced a big problem.
The problem was that the NODE_ENV value was always changed to "production" during build.
The NODE_ENV value is used by QUASAR as well as other libraries. If QUASAR forcibly changes NODE_ENV at build time, I can't make a proper build to check the development server.

NODE_ENV cannot be QUASAR-only.

Depending on the environment, it is a development server, but there are times when you need to use the production API.
Conversely, it is a built environment, but there are times when you need to use the development API.

I also updated QUASAR but faced a situation where I had to discard it.
Can you help me solve this?

**Comment by MilosPaunovic** [comment]
You can either declare your own envirnonment variable and do stuff in your code based on it, or you could set `NODE_ENV` to whatever you need [https://nodejs.dev/learn/nodejs-the-difference-between-development-and-production](https://nodejs.dev/learn/nodejs-the-difference-between-development-and-production)

**Comment by yahao87** [comment]
@MilosPaunovic Of course, I've been using it that way before.
However, if you build as you said, this value is unconditionally changed to 'production' in the QUASAR code.
What I'm saying is that you can't change this variable.

**Comment by MilosPaunovic** [comment]
`quasar build --debug` will create a build with `NODE_ENV` set to `development`.

**Comment by rstoenescu** [comment, marked as answer]
On the contrary. It must be forced because Vue, Vue Router, Pinia, Vuex, Vue I18n etc depend on specific values for NODE_ENV (it's just how they roll).

However, you can supply and use some other naming scheme in your code. Why should should you look specifically for NODE_ENV and not "SETUP_ENV" etc?
